### ISSUE TITLE
Would you be interested in adding support for the Telit LM960A18?

### ISSUE THREAD

**Issue by belusnecropolis** [bkerler/SierraWirelessGen#17]
I just recently came across your great work. I utilize the Telit radios quite a bit and I could use something like what you offer here for them. Is it possible for you to look into adding this radio as an option in your work? If you need the firmware or another program I would be glad to provide it for you.

**Comment by bkerler** [comment]
Sure, would be interesting.

**Comment by belusnecropolis** [comment]
Groovy, thank you for this.

**Comment by sand-rey** [comment]
here is four modem version of carriers
Generic
Verizon
ATT
Sprint
https://mega.nz/file/kVpxDbYC

**Comment by gsmantenna** [comment]
hope you can add Telit LM940(LN940) T77w676 in the support list

**Comment by jgsieve** [comment]
@sand-rey i'm hunting for the lm960a18 firmware, would you mind posting the latest pack if you can? thanks. As for the imei change, for a telit, there is no encrypt decrypt, i have read it's 1 comands, reading the AT commands it would be AT+CGSN=? which would return a format with the imei, update the imei and the command would be AT+CGSN=[syntax from output with new imei]. i have not tried this because i am still waiting for my unit, and looking for the latest firmware. thanks

**Comment by bkerler** [comment]
Actually there are two commands. At^NVM (lm940) and AT^NVIM (lm960). I will post the exact usage via pm if needed.

**Comment by jgsieve** [comment]
yeah, that would be great! also, would you happen to know the command to toggle 2x2 mimo? i can't find it in the AT command guide. also, would you have the latest lm960a18 firmware you could post? thanks for the response!

**Comment by jgsieve** [comment]
> Actually there are two commands. At^NVM (lm940) and AT^NVIM (lm960). I will post the exact usage via pm if needed.

i'll take you up on that offer! can you PM the details of the AT^NVIM for the lm960?

thanks!

**Comment by hrana** [comment]
Can you send any information for the LM960 to me as well? Thanks

**Comment by stefanzierpersonal** [comment]
https://gist.github.com/stefanzierpersonal/c2d3e31b035d40b0a6f7a749eb012147

**Comment by panerias** [comment]
Can you send any information for LM960 to me as well? Thank you

**Comment by bigdog3445** [comment]
Anyone have an update on this? I have this same modem. I have the Live CD working and I can't seem to get it to load my modem. Any help would be nice.

**Comment by davghz** [comment]
Did yall get it? Im currently looking at swapping it out currently

**Comment by carp4** [comment]
I was messing around with my LM960A18 and erased the IMEI with AT#NVIM=550,

I tried to write it back but I get ERROR

I also tried the modem/diag.py on LiveDVD but the file doesn’t exist. Was that a file in an older version of EDL? 

Any help to get this back is much appreciated.

**Comment by jackcrap** [comment]
> Actually there are two commands. At^NVM (lm940) and AT^NVIM (lm960). I will post the exact usage via pm if needed.

can you please PM with exact usage? I can't seem to find item number for IMEI.  Thx

**Comment by jackcrap** [comment]
> > Actually there are two commands. At^NVM (lm940) and AT^NVIM (lm960). I will post the exact usage via pm if needed.
> 
> i'll take you up on that offer! can you PM the details of the AT^NVIM for the lm960?
> 
> thanks!

Hi did bkerler get back to you with the details for the NVIM command for the lm960 and were you successful? thx

**Comment by carp4** [comment]
@jackcrap  You may find this link helpful. Lots of good info.  https://drive.google.com/drive/folders/1r6TuG1j-OFlhTeqCT8WlSTxmjsRAUF73

**Comment by jackcrap** [comment]
> @jackcrap You may find this link helpful. Lots of good info. https://drive.google.com/drive/folders/1r6TuG1j-OFlhTeqCT8WlSTxmjsRAUF73

@carp4 thank you for that very useful and informative link!!

**Comment by carp4** [comment]
@jackcrap No problem. I got a couple of the LM960A18 a while ago just to play with and mine didn't have B71. The firmware in that link will enable it. And modifying NVM has to be done with the older version of EDL which the reason for rolling back to a different date via Git. Don't try AT commands to erase NVM, I almost bricked a modem doing that. You can erase but not write back to it.

**Comment by jackcrap** [comment]
@carp4 Thanks...I was hoping the AT command would work as this modem is in my cradlepoint.  You saved me from a possible brick.  I guess I'm going to have to pull it from the cradlepoint and get an mpcie to usb to program it...if you don't mind me asking what you have to house your modem or what usb board you have?  thanks again

**Comment by carp4** [comment]
I'm not using either of my 960 modems at the moment, but I did the same thing when I was playing with them. Cheap mpcie to usb adapter off amazon to program them and then stuck them back in the router. Honestly I can't remember if my pc was booted up to Mint or Windows but I am pretty sure it was windows. I've used them in H721 and WG1602 both running rooter. I have recently been working with the Z8102AX but I don't have an adapter in that direction and haven't tried it. I probably should because where I'm at the 5G service isn't great and the LM960A18 would probably do better than my RM520N-GL.

**Comment by jackcrap** [comment]
@carp4 cool, I was hoping any of the mpcie to usb on amazon would work.  I'm looking for some 5G modems to play around with and checked the WG1602 and a few other routers out, but that Z8102AX looks pretty cool, will have to read up on that one.  Thanks for again for your help!!

**Comment by jgsieve** [comment]
Fwiw, I have changed my imei with nvm commands using EDL with the lm860a18. I had to use a pie-e native sled using Linux. It’s been a bit, but I wasn’t happy with the modem using ATT. Sent from my iPhoneOn Aug 4, 2024, at 10:45 PM, jack crap ***@***.***> wrote:﻿
@carp4 cool, I was hoping any of the mpcie to usb on amazon would work.  I'm looking for some 5G modems to play around with and checked the WG1602 and a few other routers out, but that Z8102AX looks pretty cool, will have to read up on that one.  Thanks for again for your help!!
